internal/promotion/runner/builtin/git_pusher.go

```diff
@@ -181,11 +186,27 @@ func (g *gitPushPusher) run(
 		return promotion.StepResult{Status: kargoapi.PromotionPhaseErrored},
 			fmt.Errorf("error getting last commit ID: %w", err)
 	}
+
+	// Using git provider to get commit url.
+	gpOpts := &gitprovider.Options{}
+	gitProvider, err := gitprovider.New(workTree.URL(), gpOpts)
```

Comment:
It's ok for this to be nil. The function you're calling will deal with it.

```suggestion
	gitProvider, err := gitprovider.New(workTree.URL(), nil)
```

Reply:
Although...

In other steps that interact with a git provider, there is an optional provider field that can be used to explicitly identify a provider in the cases where the provider cannot possibly be inferred from the repo URL. ex: If `https://my-git.example.com` were the base URL for my GitHub Enterprise, there's nothing in the URL to suggest the GitHub provider ought to be used. In cases such as this, a user can be explicit.

That same option should probably be added to the git pusher's config.

Reply:
Okay I'll add an optional repo URL input to the git pusher and just make it use origin (as it does now) if nothing is set.